### PR TITLE
Adds ability to track Firebase Auth UI per request

### PR DESCRIPTION
**Pull request by protocol86**
Adds internal method to FIRAuth which will be accessed within AuthUI via an extension.

### CODE REVIEW

File: Firebase/Auth/Source/FIRAuth.m [hunk 1]
```diff
@@ -1058,6 +1058,13 @@ - (void)phoneNumberSignInWithRequest:(FIRVerifyPhoneNumberRequest *)request
 }
 #endif
 
+/** @fn enableAuthUILogging
+    @brief Sets the isFirebaseAuthUIRequest flag to YES on the request configuration object.
+ */
+- (void)enableAuthUILogging {
+  _requestConfiguration.isFirebaseAuthUIRequest = YES;
```
Comment (XiangtianDai) [inline Firebase/Auth/Source/FIRAuth.m]: I'd rather make it a generic API, i.e.,

@property(nonatomic, strong) NSString *additionalFrameworkMarker;

The default value would be "FirebaseCore-iOS", and Firebase AuthUI can override it to be "FirebaseUI-iOS". This leaves room for future expansions, such as logging more information.

Also, please put this method in the `#pragma mark - Public Methods` section and treat it as if a public method in terms of threading.

**Comment by westarle** [comment]
@davidair @depoll PTAL

**Comment by XiangtianDai** [comment]
Please note that this is not a public API.

File: Firebase/Auth/Source/FIRAuth.m (hunk 1 above)
Comment (protocol86) [reply Firebase/Auth/Source/FIRAuth.m]: Done, PTAL.

**Pull request merged**